### PR TITLE
Implemented Tidal Control and Prismatic Lace

### PR DESCRIPTION
**Pull request by Zzooouhh**
Modal cost formatting for Tidal Control is taken from the Mirrodin Shard cycle, maybe there's a more elegant way to handle it? Functionally it performs good as is, though.

### CODE REVIEW

File: Mage.Sets/src/mage/cards/t/TidalControl.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        ability.addTarget(new TargetSpell(filter));
+        ability.setMayActivate(TargetController.ANY);
+        ability.addEffect(new InfoEffect("Any player may activate this ability"));
+        this.addAbility(ability);
+        ability = new SimpleActivatedAbility(Zone.BATTLEFIELD, new CounterTargetEffect(), new ManaCostsImpl("{2}"));
+        ability.addTarget(new TargetSpell(filter));
+        ability.setMayActivate(TargetController.ANY);
+        ability.addEffect(new InfoEffect("Any player may activate this ability"));
+        this.addAbility(ability);
+    }
```
Comment (emerald000) [inline Mage.Sets/src/mage/cards/t/TidalControl.java]: You can use a OrCost instead of making two seperate abilities.

**Comment by Zzooouhh** [comment]
Changed the formatting to the OrCost version.

File: Mage.Sets/src/mage/cards/t/TidalControl.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+    public TidalControl(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ENCHANTMENT}, "{1}{U}{U}");
+
+        // Cumulative upkeep-Pay {2}.
+        this.addAbility(new CumulativeUpkeepAbility(new ManaCostsImpl("{2}")));
+
+        // Pay 2 life or {2}: Counter target red or green spell. Any player may activate this ability.
+        SimpleActivatedAbility ability = new SimpleActivatedAbility(Zone.BATTLEFIELD, new CounterTargetEffect(), new OrCost(new PayLifeCost(2), new ManaCostsImpl("{2}"), "pay 2 life or pay {2}"));
+        ability.addTarget(new TargetSpell(filter));
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/t/TidalControl.java]: Did you test this successfully? I've tried implementing cards with targeted abilities that anyone can activate and couldn't make them work.

**Comment by Zzooouhh** [comment]
I've copied the implementation from Squallmonger, which is fully functional in that anybody can activate the ability. But good that you mention it, because there IS one quirk I've only noticed after revisiting it due to your comment: 
For some reason, while the player activating the ability (if it's controlled by somebody else) is paying the cost for the ability, right now it is the controller of Tidal Control that is choosing the target. I must've missed it because I didn't think of experimenting with multiple red/green spells on the stack at the same time. Any ideas how to circumvent this?

**Comment by theelk801** [comment]
That was what I encountered previously. It's something I was gonna try to fix at some point but never got around to.

**Comment by Zzooouhh** [comment]
I'll be figuring out a workaround in the meantime, but what do you suggest then? Should I let Tidal Control be dismissed for now or should I add a Todo note to it, acknowledging that while the bug exists, it generally comes up in way too fringe situations to be urgent to fix?

**Comment by theelk801** [comment]
I'd probably just leave it unimplemented for now.

**Pull request closed without merging**